src/steamship/agents/context/chathistory.py

```diff
@@ -0,0 +1,225 @@
+            )
+
+    @staticmethod
+    def get_or_create(
+        client: Client,
+        context_keys: Dict[str, str],
+        tags: List[Tag] = None,
+        initial_system_prompt: Optional[str] = None,
+    ) -> ChatHistory:
+
+        file = ChatHistory._get_existing_file(client, context_keys)
+
+        if file is None:
+            tags = tags or []
+            tags.append(Tag(kind=TagKind.DOCUMENT, name=DocTag.CHAT))
+
+            blocks = []
+            if initial_system_prompt is not None:
+                blocks.append(
+                    Block(
+                        text=initial_system_prompt,
+                        tags=[Tag(kind=TagKind.ROLE, name=RoleTag.SYSTEM)],
+                    )
+                )
+
+            file = File.create(
+                client=client,
+                blocks=blocks,
+                tags=tags,
+            )
+        return ChatHistory(file)
+
+    def append_user_message(
+        self,
+        text: str = None,
+        tags: List[Tag] = None,
+        content: Union[str, bytes] = None,
+        url: Optional[str] = None,
+        mime_type: Optional[MimeTypes] = None,
+    ) -> Block:
+        """Append a new block to this with content provided by the end-user."""
+        tags = tags or []
+        tags.append(
+            Tag(
+                kind=TagKind.CHAT, name=ChatTag.ROLE, value={TagValueKey.STRING_VALUE: RoleTag.USER}
+            )
+        )
+        return self.file.append_block(
+            text=text, tags=tags, content=content, url=url, mime_type=mime_type
+        )
+
+    def append_system_message(
+        self,
+        text: str = None,
+        tags: List[Tag] = None,
+        content: Union[str, bytes] = None,
+        url: Optional[str] = None,
+        mime_type: Optional[MimeTypes] = None,
+    ) -> Block:
+        """Append a new block to this with content provided by the system, i.e., instructions to the assistant."""
+        tags = tags or []
+        tags.append(
+            Tag(
+                kind=TagKind.CHAT,
+                name=ChatTag.ROLE,
+                value={TagValueKey.STRING_VALUE: RoleTag.SYSTEM},
+            )
+        )
+        return self.file.append_block(
+            text=text, tags=tags, content=content, url=url, mime_type=mime_type
+        )
+
+    def append_agent_message(
+        self,
+        text: str = None,
+        tags: List[Tag] = None,
+        content: Union[str, bytes] = None,
+        url: Optional[str] = None,
+        mime_type: Optional[MimeTypes] = None,
+    ) -> Block:
+        """Append a new block to this with content provided by the agent, i.e., results from the assistant."""
+        tags = tags or []
+        tags.append(
+            Tag(
+                kind=TagKind.CHAT,
+                name=ChatTag.ROLE,
+                value={TagValueKey.STRING_VALUE: RoleTag.ASSISTANT},
+            )
+        )
+        return self.file.append_block(
```

Comment:
nit: some code duplication here. Would wrap this into _append_message(role="") method